backwardcompatibilityml/widget/compatibility_analysis.py

```diff
@@ -149,51 +214,19 @@ def __init__(self, folder_name, number_of_epochs, h1, h2, training_set, test_set
 
-        resource_package = __name__
-        javascript_path = '/'.join(('resources', 'widget-build.js'))
-        css_path = '/'.join(('resources', 'widget.css'))
-        html_template_path = '/'.join(('resources', 'widget.html'))
-        widget_javascript = pkg_resources.resource_string(
-            resource_package, javascript_path).decode("utf-8")
-        widget_css = pkg_resources.resource_string(
-            resource_package, css_path).decode("utf-8")
-        app_html_template_string = pkg_resources.resource_string(
-            resource_package, html_template_path).decode("utf-8")
-
-        api_service_environment = build_environment_params(self.flask_service.env)
-        api_service_environment["port"] = self.flask_service.port
-
-        app_html_template = Template(app_html_template_string)
-        html_string = app_html_template.render(
-            widget_css=widget_css,
-            widget_javascript=widget_javascript,
-            api_service_environment=json.dumps(api_service_environment),
-            data=json.dumps(None))
-
-        self.html_widget = HTML(html_string)
-        display(self.html_widget)
-
-        @FlaskHelper.app.route("/api/v1/start_sweep", methods=["POST"])
-        def start_sweep():
-            self.sweep_manager.start_sweep()
-            return {
-                "running": self.sweep_manager.sweep_thread.is_alive(),
-                "percent_complete": self.sweep_manager.get_sweep_status()
-            }
-
-        @FlaskHelper.app.route("/api/v1/sweep_status", methods=["GET"])
-        def get_sweep_status():
-            return {
-                "running": self.sweep_manager.sweep_thread.is_alive(),
-                "percent_complete": self.sweep_manager.get_sweep_status()
-            }
-
-        @FlaskHelper.app.route("/api/v1/sweep_summary", methods=["GET"])
-        def get_data():
-            return Response(
-                json.dumps(self.sweep_manager.get_sweep_summary()),
-                mimetype="application/json")
-
-        @FlaskHelper.app.route("/api/v1/evaluation_data/<int:evaluation_id>")
-        def get_evaluation(evaluation_id):
-            return self.sweep_manager.get_evaluation(evaluation_id)
+        if dev_mode:
+            # Run Flask outside the FlaskHelper wrapper
+            # HTML will be served by webpack server
+            from flask import Flask
+            from flask_cors import CORS
+            self.flask_service = Flask(__name__)
+            CORS(self.flask_service)
+        else:
+            self.flask_service = FlaskHelper(ip="0.0.0.0", port=port)
+            api_service_environment = build_environment_params(self.flask_service.env)
+            api_service_environment["port"] = self.flask_service.port
+            html_string = render_widget_html(api_service_environment)
+            self.html_widget = HTML(html_string)
+            display(self.html_widget)
+
+        init_app_routes(self.flask_service, self.sweep_manager)
```

Comment:
Note that in non-dev mode, this actually has to initialize the routes like so:

`init_app_routes(FlaskHelper.app, self.sweep_manager)`

I might have missed that before, if so I apologize.

Reply:
Good catch, thanks!